bgpd/bgp_fsm.c

```diff
@@ -2641,7 +2647,7 @@ int bgp_event_update(struct peer *peer, enum bgp_fsm_events event)
 		 * we need to indicate that the peer was stopped in the return
 		 * code.
 		 */
-		if (!dyn_nbr && !passive_conn && peer->bgp) {
+		if (!dyn_nbr && !passive_conn && peer->bgp && ret != -2) {
```

Comment:
I suggest giving a name for `-2` (aka `#define`).

Reply:
changed to an enum in a later commit.  Since the whole thing needed to be updated